### ISSUE TITLE
Server-Side-Pagination with Composition API

### ISSUE THREAD

**Issue by anli-xsigns** [feathersjs-ecosystem/feathers-vuex#432]
Hello,
I try to get Server-Side-Pagination with the Composition API up and running. The component I use is a quasar table (https://quasar.dev/vue-components/table#Server-side-pagination%2C-filter-and-sorting).

```
setup(props, context) {

      const { Buildings } = context.root.$FeathersVuex.api;

      const pagination = reactive({
        sortBy: 'name',
        descending: false,
        page: 1,
        rowsPerPage: 10,
        rowsNumber: 0
      });

      const queryParams = computed(() => {
        return {
          query: {
            $sort: { name: 1 },
            $limit: pagination.rowsPerPage,
            $skip: 0
          },
          paginate: true
        }
      });

      const { items: buildings, paginationData } = useFind({
        model: Buildings,
        params: queryParams
      });

      function onRequest(requestProps) {
        const { rowsPerPage } = requestProps.pagination;
        pagination.rowsPerPage = rowsPerPage;
      }

      return {
        buildings,
        hasLoaded,
        pagination,
        queryParams,
      }
    }
```
In https://vuex.feathersjs.com/composition-api.html#options it states "Set params.paginate to true to turn off realtime updates for the results and defer pagination to the API server. Pagination works the same as for the makeFindMixin pagination.".  However, I don't want to turn off realtime updates -> if records come in that match the currently shown page these should replace the current page.
If I set paginate to true, no rows are fetched from the server with the above setup.

I have three questions:
- What am I missing to get the above code up and running and keep getting realtime updates?
- How can I initiate the first request?
- How can I get the total number of rows after the request has finished? I need to set pagination.rowsNumber and there is no callback or promise for paginationData to decide when to receive the information from paginationData (already saw total in there).

Thanks for your help in advance!

**Comment by marshallswain** [comment]
I believe that https://github.com/feathersjs-ecosystem/feathers-vuex/issues/430 is probably the biggest problem.  I'm just fixing this, now.  I'll revisit this after I deploy the fix.

**Comment by marshallswain** [comment]
> I don't want to turn off realtime updates -> if records come in that match the currently shown page these should replace the current page.

The only simple way to solve this problem is to rerun the exact same query against the api server.

Another way to solve this would be to run some algorithm against every new record that would check new records against each page of records in the pagination.  It's quite a tedious task.  

If you think of another solution, please let me know.  In the meantime, fixing that other issue will get us up and running with server pagination.

**Comment by marshallswain** [comment]
The bug where records would not show when `paginate: true` was turned on is now fixed in `feathers-vuex@3.7.0`.

**Comment by marshallswain** [comment]
You might take a look at the new `FeathersVuexPagination` component, which you can use with the `useFind` Composition API utility: https://vuex.feathersjs.com/composition-api.html#feathersvuexpagination

**Comment by olegdev** [comment]
Hey! I integrate an ant-design table with feathers pagination and i need a `total` of items too.
Any help here?

**Comment by olegdev** [comment]
I found solution like:
`:total="latestQuery && latestQuery.response ? latestQuery.response.total : 0"`

**Comment by marshallswain** [comment]
@olegdev That is exactly the way it was intended to be used.  Thank you for sharing.  I'm going to close up this issue as it seems to be working after the fixes deployed in version 3.7.  

The only feature left in this issue that's not fully implemented is 100% realtime support.  Realtime updates are supported for individual records, so we still get realtime support that way.  However, keeping paginated lists realtime is a bigger scope than I'm willing to take on right now.  I still haven't even found a logical solution that would be accurate in every use case.
